cloud-function/src/handlers/proxy-content.ts

```diff
@@ -36,12 +38,22 @@ export const proxyContent = createProxyMiddleware({
             res.statusCode = 200;
             res.setHeader("Content-Type", "text/html");
             return Buffer.from(await tryHtml.arrayBuffer());
-          } else if (!notFoundBuffer) {
-            const response = await fetch(`${target}${NOT_FOUND_PATH}`);
-            notFoundBuffer = await response.arrayBuffer();
+          } else if (!notFoundBufferHtml) {
```

Comment:
Either, make intent a bit clearer with:

```suggestion
          }
          
          if (!notFoundBufferHtml || !notFoundBufferJson) {
```

Or make worse-case performance marginally better with (pseudocode):
```
if url.endsWith(json) {
  if !notFoundBufferJson {
    fetch and set it
  }
  return notFoundBufferJson
}

if !notFoundBufferHtml {
  fetch and set it
}
return notFoundBufferHtml
```

Reply:
Fixed in 9366b6f8d8a95dede71e9573c74aa714f3993508